Table commit bug

tungi52 opened the pull request:
In #826 there is an interesting bug described. I made this PR draft, because the problem consist of several parts. First I noticed, that the `MudFormComponent `never removed itself from a form, therefore the `Remove `method of the `TableRowValidator `never called. So references stucked in the dictionary. The second problem, that `TableRowValidator ` uses strong references to `MudFormComponent`, so this prevents the components to be disposed. So the number of references are grows and grows now.
I make a fix, this works fine, the components GC-d now.
I have two questions which are considareble:
- This issue maybe affects `MudForm`. Should we fix it too?
- If there is a row validation error, I think we should prevent swiching row until the error fixed

henon commented:
@tungi52 first of all, wow, good catch on the memory leak!

> * This issue maybe affects `MudForm`. Should we fix it too?

Definitely, and we seriously need a test for that.

> * If there is a row validation error, I think we should prevent swiching row until the error fixed

Sounds reasonable to me.

henon commented on src/MudBlazor/Base/MudFormComponent.cs:
```diff
@@ -491,7 +491,7 @@ protected virtual void Dispose(bool disposing)
 
         void IDisposable.Dispose()
         {
-            //ParentForm?.Remove(this);
+            Form?.Remove(this);
             DetachValidationStateChangedListener();
             Dispose(disposing: true);
         }
```
I think it was commented out because it failed in some special case. Please wrap it with `try { Form?.Remove(this); } catch { /* ignore */ }`

henon reviewed (changes requested):
Btw, what exactly needs fixing in MudForm?

tungi52 commented:
@henon, Nothing need to change in `MudForm`. I just saw that they use the same interface, and `Remove` is never called so I thought there could be the same issue there. I made a lot of tests, and if someone doesn't use `MudForm` entirely wrong, this memory leak will never happen. I also followed back when this `Remove` method call commented out. It is a very old change, so just to be sure not breaking anything, I commented the inside of the `Remove` method of  `MudForm`.
I changed the code, it seems I had luck with GC for the first time, so I drop the `WeakReferences` and added a `RemoveAll` method to `IForm` interface.
I made a simple test too, however I'm not proud of it.

tungi52 commented:
Ok, no need the RemoveAll. I find the disposal problem in `MudDebouncedInput`.

henon commented:
Thanks Andras!

henon commented:
The test is perfectly fine. Tests should be stupid, not sophisticated. ;)
